src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -424,6 +427,17 @@ public AzureVMDeploymentInfo createDeployment(
         }
     }
 
+    private boolean checkImageParameter(AzureVMAgentTemplate template) {
+        if (StringUtils.isBlank(template.getImagePublisher())
+                || StringUtils.isBlank(template.getImageOffer())
+                || StringUtils.isBlank(template.getImageSku())
+                || StringUtils.isBlank(template.getImageVersion())) {
+            LOGGER.warning("Missing Image Reference information when trying to add purchase plan to ARM template");
```

Comment:
if use custom image, warning message is ok? should be error, right?

Reply:
got it